### ISSUE TITLE
Plugin is removed for all imports after skipSelf

### ISSUE THREAD

**Issue by charlag** [PepsRyuu/nollup#162]
Hi
I stumbled upon a bug in nollup.
When there are multiple inputs, context is only valid for the first one because of this line:
https://github.com/PepsRyuu/nollup/blob/1979e73db8f4f18bf1025212e9e857942f226105/lib/impl/PluginContext.js#L97

As it is a global context, it is modified forever.
This is easy to reproduce, I created a minimal repo:
```sh
git clone git@github.com:tutao/nollup-bug.git && cd nollup-bug && npm ci && node make
```
commonjs plugin calls `resolveId` with `skipSelf` which causes the line above to remove it from context. For the second input commonjs won't be invoked, module won't be transformed to ES module and `bluebird` will be undefined (probably module is processed twice and overwritten?).
If you commend out the second input then it works.

I would love to send a PR but I'm not confident enough with internals, I hope this helps.
Thanks for nollup!

**Comment by PepsRyuu** [comment]
Repository doesn't exist apparently. ☹

**Comment by charlag** [comment]
Oh, sorry, since when does GH default to private repos?
Should be there now

**Comment by PepsRyuu** [comment]
Fixed in ```0.14.2```. Would appreciate it if you could confirm. :)

**Comment by charlag** [comment]
It seems to be solved, thank you a lot!
